# Cleaning AirBnB data using python for SQL project

Author: S M Maisoon Ul Amin

Codebase: https://github.com/maisoonaminongithub

Portfolio: www.maisoonamin.com

## Key takeways 

1. I have downloaded scapped dataset from a service called https://insideairbnb.com/get-the-data The dataset contains data for toronto airbnb, February 2024.
2. Many columns have missing values, particularly in textual descriptions and detailed host information. Some numerical fields like host response rate are stored as strings and might need conversion for analysis.
3. 

## About dataset
The dataset consists of 20,630 entries and 75 columns, capturing various aspects of Airbnb listings. Here's a summary of the key points:

1. Basic Information: Each listing has an ID, a URL, and details such as the name, description, and photos.
2. Host Information: Data on hosts includes IDs, names, response times, superhost status, and the number of listings they have.
3. Location Details: Includes URLs to neighborhood overviews and specific location data, although many of these fields have missing values.
4. Availability and Booking: Information on price, minimum and maximum nights for bookings, and availability over different time spans.
5. Review Scores: Various metrics on reviews, such as accuracy, cleanliness, and overall rating.


## Loading the dataset

In [1]:
# analysing the dataset using pandas
import pandas as pd

# Load the dataset
file_path = '/kaggle/input/airbnb-toronto-022024/airbnb-listingData-24022024.csv'
data = pd.read_csv(file_path)

## Find missing data for the whole set and delete them if not neccesary

In [2]:
# Calculate the percentage of missing data for each column
missing_data = data.isnull().mean() * 100
missing_data.sort_values(ascending=False)

neighbourhood_group_cleansed    100.000000
calendar_updated                100.000000
host_neighbourhood               58.768783
license                          53.645177
host_about                       52.840524
                                   ...    
latitude                          0.000000
availability_30                   0.000000
availability_60                   0.000000
availability_90                   0.000000
id                                0.000000
Length: 75, dtype: float64

### Here are some columns in the dataset with significant amounts of missing data:

1. neighbourhood_group_cleansed and calendar_updated: Both have 100% missing data, indicating that these fields are either unused or not applicable to this subset of listings.
2. host_neighbourhood: Approximately 58.77% missing, suggesting many listings do not specify this detail.
3. license: About 53.65% missing, which may reflect varying requirements for a license depending on local regulations or incomplete data entry.
4. host_about: Missing in 52.83% of the rows, indicating many hosts have not provided a description of themselves.

### Finding coloumns with missing data and list them



In [3]:
# Find the percentage of missing data for each column and filter out the ones that have any missing data
columns_with_missing_data = data.isnull().mean() * 100
columns_with_missing_data = columns_with_missing_data[columns_with_missing_data > 0].sort_values(ascending=False)
columns_with_missing_data


neighbourhood_group_cleansed    100.000000
calendar_updated                100.000000
host_neighbourhood               58.768783
license                          53.645177
host_about                       52.840524
neighborhood_overview            45.249636
neighbourhood                    45.244789
host_response_time               33.160446
host_response_rate               33.160446
review_scores_location           26.345128
host_acceptance_rate             26.340281
review_scores_communication      26.335434
review_scores_checkin            26.335434
review_scores_accuracy           26.335434
last_review                      26.335434
first_review                     26.335434
review_scores_value              26.335434
reviews_per_month                26.335434
review_scores_cleanliness        26.325739
review_scores_rating             26.320892
price                            25.773146
beds                             25.584101
bathrooms                        25.458071
host_locati

### Delete all coloumns with more then 26% missing data - this is because based on our business objective, we actually dont need them.

In [4]:
# Columns with more than 26% missing data
columns_to_delete = columns_with_missing_data[columns_with_missing_data > 26].index

# Drop these columns from the dataset
data_cleaned_further = data.drop(columns=columns_to_delete)

data.columns

Index(['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name',
       'description', 'neighborhood_overview', 'picture_url', 'host_id',
       'host_url', 'host_name', 'host_since', 'host_location', 'host_about',
       'host_response_time', 'host_response_rate', 'host_acceptance_rate',
       'host_is_superhost', 'host_thumbnail_url', 'host_picture_url',
       'host_neighbourhood', 'host_listings_count',
       'host_total_listings_count', 'host_verifications',
       'host_has_profile_pic', 'host_identity_verified', 'neighbourhood',
       'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude',
       'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms',
       'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'minimum_minimum_nights',
       'maximum_minimum_nights', 'minimum_maximum_nights',
       'maximum_maximum_nights', 'minimum_nights_avg_ntm',
       'maximum_nights_avg_ntm', 'ca

In [5]:
# Save the DataFrame to a CSV file
data.to_csv('airbnb_listings_cleaned.csv', index=False)

# Output the file path to check it
print("The CSV file has been saved successfully at:")
print('/kaggle/working/airbnb_listings_cleaned.csv')


The CSV file has been saved successfully at:
/kaggle/working/airbnb_listings_cleaned.csv
